packages/opentelemetry-metrics/src/SumObserverMetric.ts

```diff
@@ -30,23 +29,35 @@ export class SumObserverMetric
   constructor(
     name: string,
     options: api.MetricOptions,
-    batcher: Batcher,
+    processor: Processor,
     resource: Resource,
     instrumentationLibrary: InstrumentationLibrary,
     callback?: (observerResult: api.ObserverResult) => unknown
   ) {
     super(
       name,
       options,
-      batcher,
+      processor,
       resource,
       MetricKind.SUM_OBSERVER,
       instrumentationLibrary,
       callback
     );
   }
 
-  protected createObserverResult(): ObserverResult {
-    return new MonotonicObserverResult();
+  protected _processResults(observerResult: ObserverResult) {
+    observerResult.values.forEach((value, labels) => {
+      const instrument = this.bind(labels);
+      // SumObserver is monotonic which means it should only accept values
+      // greater or equal then previous value
+      const previous = instrument.getAggregator().toPoint();
+      let previousValue = -Infinity;
+      if (previous.timestamp[0] !== 0 || previous.timestamp[1] !== 0) {
+        previousValue = previous.value as LastValue;
+      }
+      if (value >= previousValue) {
```

Comment:
should we warn when an user tries when its not the case ? I mean the user have no way to know that they are doing it wrong currently

Reply:
I would avoid warning, as this is monotonic metric - SumObserver is monotonic, so it does what it was designed to do. Spec doesn't say anything about warning user as it is the desired behaviour. We can ask in spec if there are any plans like this, but for now I would avoid it. The spec might still change yet, but I would avoid adding this in this PR now.